ServerCore/Areas/Identity/Pages/Account/Edit.cshtml.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+
+namespace ServerCore.Areas.Identity.Pages.Account
+{
+    [AllowAnonymous]
+    public class EditModel : PageModel
+    {
+        private readonly UserManager<IdentityUser> _userManager;
+        private readonly PuzzleServerContext _context;
+
+        public EditModel(
+            UserManager<IdentityUser> userManager,
+            PuzzleServerContext context)
+        {
+            _userManager = userManager;
+            _context = context;
+        }
+
+        [BindProperty]
+        public PuzzleUser PuzzleUser { get; set; }
+
+        public string ReturnUrl { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int userId, string returnUrl = null)
+        {
+            var thisPuzzleUser = await PuzzleUser.GetPuzzleUserForCurrentUser(_context, User, _userManager);
```

Comment:
If we always redirect the user to the built-in settings page that simplifies these logic checks quite a bit since it will only allow global admins :)

Reply:
Yep, I'll change the policy attribute and then remove all the checks.